### ISSUE TITLE
Is Storing Access_Token in Server-Side Cookies a Best Practice for API Validation?

### ISSUE THREAD

**Issue by zuko-firelord** [authorizerdev/authorizer#463]
I am using Next.js 15 for the frontend and a Go backend. While reviewing the example of auth-nextjs, I noticed that the access_token is stored in a server-side cookie that can be use for validating API requests in middleware.
![image](https://github.com/user-attachments/assets/a3ada7d9-4851-40fe-9199-1ea661cf0d3b)

Is this approach secure and recommended?
If I decide not to store the access_token on server-side cookie, can I validate API requests using a session_token instead?

**Comment by zuko-firelord** [comment]
@lakhansamani

**Comment by lakhansamani** [comment]
@zuko-firelord yes u can also validate it using session_token, but make sure that cookie is present.

**Comment by zuko-firelord** [comment]
@lakhansamani i couldnt find any function in authorizer-go related to validating APIs using the session_token. could you let me know the name of the function?

**Comment by zuko-firelord** [comment]
i want to raise a PR for an example of authorizer-nextjs15 with middleware functionality and implement the backend authorize API. im not good at frontend stuff, but i'll give it a shot. Where should I push my code?

**Comment by lakhansamani** [comment]
@zuko-firelord  thanks
you can create PR here: https://github.com/authorizerdev/examples

**Comment by zuko-firelord** [comment]
> @zuko-firelord thanks you can create PR here: https://github.com/authorizerdev/examples

ive raised a pr...please cross-check that logics are correct, did it very short amt of time...pardon me for ui

**Comment by zuko-firelord** [comment]
> @lakhansamani i couldnt find any function in authorizer-go related to validating APIs using the session_token. could you let me know the name of the function?

?

**Comment by lakhansamani** [comment]
@zuko-firelord  there is API
https://docs.authorizer.dev/core/graphql-api#validate_session

However this needs cookie and will return new session for security reasons.
